src/pages/index.js

```diff
@@ -84,26 +104,153 @@ editCardPopup.setEventListeners();
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
+/*Delete card confirmation*/
+
+const deleteConfirmPopup = new PopupWithConfirmation({
+  popupSelector: "#modal-delete-card",
+  handleFormSubmit: () => {},
+});
+deleteConfirmPopup.setEventListeners();
+
+/*Edit Avatar*/
+
+const avatarProfilePopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  avatarSubmit
+);
+avatarProfilePopup.setEventListeners();
+
+const EditAvatarBtn = document.querySelector(".profile__avatar-button");
+EditAvatarBtn.addEventListener("click", () => {
+  avatarProfilePopup.open();
+});
 /*  Event Handlers */
 
-function handleProfileEditSubmit(data) {
+function avatarSubmit(link) {
+  avatarProfilePopup.modalLoading(true);
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      console.log(res);
+      userInfo.updateProfileImage(res.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarProfilePopup.modalLoading(false);
+    });
+  avatarProfilePopup.close();
+  avatarFormElement.reset();
+}
+
+function handleDeleteCard(card) {
+  //console.log("Delete button clicked");
+  deleteConfirmPopup.open();
+
+  deleteConfirmPopup.confirmDelete(() => {
+    // console.log("Deletion confirmed");
+    api
+
+      .deleteCard(card.getId())
+      .then(() => {
+        // console.log("Card deletion API succeeded");
+        card.removeCard();
+        deleteConfirmPopup.close();
+      })
+      .catch((err) => {
+        console.error("Card deletion API failed:", err);
+      });
+  });
+}
+
+function handleLikeIcon(card) {
+  if (card.getIsLiked()) {
+    api
+      .removeLike(card.getId())
+      .then(() => {
+        card.handleLikeCard(false);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+        // Optionally revert the UI state if there's an error
+      });
+  } else {
+    api
+      .addLike(card.getId())
+      .then(() => {
+        card.handleLikeCard(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+        // Optionally revert the UI state if there's an error
+      });
+  }
+}
+
+/*function handleProfileEditSubmit(data) {
   userInfo.setUserInfo(data);
   editCardPopup.close();
```

Comment:
`editCardPopup` should be named `profilePopup` because it has nothing to do with `cards`